content/blog/managing-aws-credentials-on-cicd/index.md

```diff
@@ -0,0 +1,231 @@
+
+## Automate Rotating and Revoking those Credentials
+
+When a CI/CD system has access to an IAM User’s AWS credentials, we need to take steps to rotate those credentials regularly, even if we trust the CI/CD provider to secure them properly.
+
+> This is one of the advantages of running your CI/CD system within your AWS account. If your deployment workers are simply EC2 instances, you wouldn’t need to share IAM User credentials. The EC2 instances themselves [can be configured to assume an IAM Role](https://aws.amazon.com/blogs/security/new-attach-an-aws-iam-role-to-an-existing-amazon-ec2-instance-by-using-the-aws-cli/).
+>
+> The details about how IAM Roles are a more secure alternative to IAM User credentials are covered in the next section.
+
+“Just create a new IAM User credential, log into the CI/CD system, and then update the job’s settings” sounds simple enough. But it’s likely going to be tedious, and if you need to repeat the process for dozens if not hundreds of CI/CD pipelines, you need a better solution. Unless you start with an automated approach for key rotation, you are unlikely to add one later.
+
+The AWS security blog describes how to [rotate access keys for IAM Users](https://aws.amazon.com/blogs/security/how-to-rotate-access-keys-for-iam-users/).
+
+1. Create a second access key in addition to the one in use.
+1. Update all your applications to use the new access key and validate that the applications are working.
+1. Change the state of the previous access key to inactive.
+1. Validate that your applications are still working as expected.
+1. Delete the inactive access key.
+
+## During deployments, exchange the IAM user credentials for a more privileged IAM role
+
+Now that we have a regularly rotated, low-privileged credential to use during deployments, the next step is to use “assume role” to exchange those IAM User credentials for an IAM Role credential. This sounds confusing at first. Why exchange one credential for another? Why not just use the same credentials? In other words, what is an IAM Role, and why is this so much better than an IAM User? The credentials used to assume an IAM Role are only valid for a short period (minutes to hours). After a fixed duration, there is no threat if disclosed.
+
+Earlier, we mitigate potentially exposing IAM User credentials by using our CI/CD provider’s ability to store secret values and automatically rotating credentials. Using the IAM Role credentials adds another layer of security, but within the context of the machine itself during the deployment.
+
+So how do you use these credentials? The AWS Secure Token Service can do this, but in practice, you’ll want to use a lightweight tool to do this.
+
+### Configuring AWS Security Group
```

Comment:
Perhaps this was added by mistake? I don't think this ("AWS security groups") applies to this post.

Instead we'd probably want to explain how the AWS CLI tool can be used to assume a role, (e.g. https://aws.amazon.com/premiumsupport/knowledge-center/iam-assume-role-cli/). (Another tool that is perhaps easier to the task is the https://github.com/remind101/assume-role tool.)

So that when you go to deploy your Pulumi application, you are doing so as the "higher-privledge IAM Role" rather than the "no access IAM User account".

Reply:
deleted